### PR TITLE
Add docs for storage class mapping

### PR DESCRIPTION
**Pull request by xinyanw409**
Add documentation for storage class mapping.

Signed-off-by: wxinyan <wxinyan@wxinyan-a01.vmware.com>

### CODE REVIEW

File: docs/guest.md [hunk 1]
```diff
@@ -184,4 +184,10 @@ CloneFromSnapshot CRD has some key phases for the `.status.phase` field:
 * New: clone from snapshot is not completed
 * InProgress: the vSphere volume snapshot is being downloaded from remote repository.
 * Completed: clone from snapshot is completed
-* Failed: clone from snapshot is failed
\ No newline at end of file
+* Failed: clone from snapshot is failed
+
+### StorageClass Mapping
+Velero plugin for vsphere v1.1.0 and above supports changing the storage class persistent volumes and persistent volume claims during restores,
```
Comment (xing-yang) [inline docs/guest.md:190]: the storage class persistent volumes and persistent volume claims -> the storage class of persistent volumes

File: docs/vanilla.md [hunk 2]
```diff
@@ -305,4 +306,45 @@ Download CRD has some key phases for the `.status.phase` field:
 * InProgress: download is in progress
 * Completed: download is completed
 * Retry: download is retried. When there is any failure during the download of backup data, download will be retried
-* Failed: download is failed
\ No newline at end of file
+* Failed: download is failed
+
+### StorageClass Mapping
```
Comment (xing-yang) [inline docs/vanilla.md:311]: make this a standalone doc

File: docs/vanilla.md [hunk 3]
```diff
@@ -305,4 +306,45 @@ Download CRD has some key phases for the `.status.phase` field:
...
+data:
+  # add 1+ key-value pairs here, where the key is the old
+  # storage class name and the value is the new storage
+  # class name.
+  <old-storage-class>: <new-storage-class>
+```
+
+Storage class is required for restore. If no storage class is specified in the PVC during backup, user can specify 
+"com.vmware.cnsdp.emptystorageclass" as the old storage class name to map to a new existing storage class name at restore time.
+```bash
```
Comment (xing-yang) [inline docs/vanilla.md:343]: add empty line before bash

File: docs/vanilla.md [hunk 4]
```diff
@@ -305,4 +306,45 @@ Download CRD has some key phases for the `.status.phase` field:
...
-* Failed: download is failed
\ No newline at end of file
+* Failed: download is failed
+
+### StorageClass Mapping
+Velero plugin for vsphere v1.1.0 and above supports changing the storage class persistent volumes and persistent volume claims during restores, 
+by configuring a Storage Class mapping in config map in the Velero namespace.
+
+Here is an example for ConfigMap in YAML, which contains <old-storage-class>:<new-storage-class> mapping pairs.
+```bash
```
Comment (xing-yang) [inline docs/vanilla.md:316]: add empty line before bash

**Comment by svcbot-qecnsdp** [comment]
Congratulations !! 

All the pre-check tests passed
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/442/

File: docs/vanilla.md [hunk 5]
```diff
@@ -305,4 +306,45 @@ Download CRD has some key phases for the `.status.phase` field:
 * InProgress: download is in progress
 * Completed: download is completed
 * Retry: download is retried. When there is any failure during the download of backup data, download will be retried
-* Failed: download is failed
\ No newline at end of file
+* Failed: download is failed
+
+### StorageClass Mapping
+Velero plugin for vsphere v1.1.0 and above supports changing the storage class persistent volumes and persistent volume claims during restores,
```
Comment (lintongj) [inline docs/vanilla.md:312]: Add a link to Velero document, https://velero.io/docs/v1.5/restore-reference/

File: docs/vanilla.md (hunk 2 above)
Comment (lintongj) [reply docs/vanilla.md:311]: Agree with Xing. Please keep this as a standalone doc. 

StorageClass Mapping is an option of restore, so it is not good to create a new section that is parallel with the `Restore` section. Instead, you can add a reference before the following line in vanilla.md.
```
For more restore options, please refer to [Velero Document](https://velero.io/docs/v1.5/).
```
For example,
```
For the volume restores with changing storage class, please refer to <the link to your storage class mapping doc>
```

Comment (xinyanw409) [reply docs/vanilla.md]: Moved storage class mapping part to a standalone doc.

**Review by lintongj — APPROVED** [review]
/lgtm

**Comment by svcbot-qecnsdp** [comment]
Congratulations !! 

All the pre-check tests passed
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/443/

**Pull request merged**